src/cript/nodes/subobjects/computation_forcefield.py

```diff
@@ -0,0 +1,127 @@
+        implicit_solvent: str = "",
+        source: str = "",
+        description: str = "",
+        data: Union[Data, None] = None,
+        citation: Union[List[Citation], None] = None,
+        **kwargs
+    ):
+        if citation is None:
+            citation = []
+        super().__init__("ComputationForcefield")
+
+        self._json_attrs = replace(
+            self._json_attrs,
+            key=key,
+            building_block=building_block,
+            coarse_grained_mapping=coarse_grained_mapping,
+            implicit_solvent=implicit_solvent,
+            source=source,
+            description=description,
+            data=data,
+            citation=citation,
+        )
+        self.validate()
+
+    @property
+    def key(self) -> str:
+        return self._json_attrs.key
+
+    @key.setter
+    def key(self, new_key: str):
+        new_attrs = replace(self._json_attrs, key=new_key)
+        self._update_json_attrs_if_valid(new_attrs)
+
+    @property
+    def building_block(self) -> str:
+        return self._json_attrs.building_block
+
+    @building_block.setter
+    def building_block(self, new_building_block: str):
+        new_attrs = replace(self._json_attrs, building_block=new_building_block)
+        self._update_json_attrs_if_valid(new_attrs)
+
+    @property
+    def coarse_grained_mapping(self) -> str:
+        return self._json_attrs.coarse_grained_mapping
+
+    @coarse_grained_mapping.setter
+    def coarse_grained_mapping(self, new_coarse_grained_mapping: str):
+        new_attrs = replace(self._json_attrs, coarse_grained_mapping=new_coarse_grained_mapping)
+        self._update_json_attrs_if_valid(new_attrs)
+
+    @property
+    def implicit_solvent(self) -> str:
+        return self._json_attrs.implicit_solvent
+
+    @implicit_solvent.setter
+    def implicit_solvent(self, new_implicit_solvent: str):
+        new_attrs = replace(self._json_attrs, implicit_solvent=new_implicit_solvent)
+        self._update_json_attrs_if_valid(new_attrs)
+
+    @property
+    def source(self) -> str:
+        return self._json_attrs.source
+
+    @source.setter
+    def source(self, new_source: str):
+        new_attrs = replace(self._json_attrs, source=new_source)
+        self._update_json_attrs_if_valid(new_attrs)
+
+    @property
+    def description(self) -> str:
+        return self._json_attrs.description
+
+    @description.setter
+    def description(self, new_description: str):
+        new_attrs = replace(self._json_attrs, description=new_description)
+        self._update_json_attrs_if_valid(new_attrs)
+
+    @property
+    def data(self) -> Union[Data, None]:
+        return self._json_attrs.data
+
+    @data.setter
+    def data(self, new_data: Union[Data, None]):
+        new_attrs = replace(self._json_attrs, data=new_data)
+        self._update_json_attrs_if_valid(new_attrs)
+
+    @property
+    def citation(self) -> List[Citation]:
+        return self._json_attrs.citation.copy()
```

Comment:
why returning the copy instead of the attribute?

Reply:
citation is a list. and python returns a shallow copy of lists, so you would be able to change the members of the original list directly.
This is prevented by returning a copy of the list, so any modification on the copy doesn't change the node list.

For encapsulation of the list, this is necessary.

Reply:
got it! You are 100% right, this should not have been a comment